server/src/main/java/io/druid/metadata/IndexerSQLMetadataStorageCoordinator.java

```diff
@@ -85,27 +89,48 @@ public void start()
     connector.createSegmentTable();
   }
 
+  @Override
   public List<DataSegment> getUsedSegmentsForInterval(
       final String dataSource,
       final Interval interval
   ) throws IOException
   {
+    return getUsedSegmentsForIntervals(dataSource, ImmutableList.of(interval));
+  }
+
+  @Override
+  public List<DataSegment> getUsedSegmentsForIntervals(
```

Comment:
This could potentially return the same segments multiple times, I think. Is that bad?